src/Compilers/Core/Portable/PEWriter/PeWriter.cs

```diff
@@ -196,6 +199,14 @@ internal static class PeWriter
                 debugDirectoryBuilder = null;
             }
 
+            var strongNameProvider = context.Module.CommonCompilation.Options.StrongNameProvider;
+
+            var corFlags = properties.CorFlags;
+            if (privateKeyOpt != null && strongNameProvider.Capability == SigningCapability.SignsPeBuilder)
```

Comment:
Is `strongNameProvider.Capability == SigningCapability.SignsPeBuilder` ever false? If not, consider asserting instead.

Reply:
Yes, `DesktopStrongNameProvider` has `SigningCapability.SignsStream`.

Reply:
It looks like we're only passing private key when `SignsPeBuilder` though.

Reply:
Ah yep, I see what you mean.  I've removed another instance of this further down in the file.